### PR TITLE
FEAT: Add initial schema

### PR DESCRIPTION
**Pull request by mariusmihaic**
Added initial `BlockResult` avro schema w.r.t (see [this structures](https://github.com/ElrondNetwork/elastic-indexer-go/tree/feat/new-indexer/data) ):
- `Block`
- `Transactions`
- `SCResults`
- `Receipts`
- `Logs`
- `StateChanges`

Also added a MakeFile which executes a go app (codegen.go), which takes as **input**: avro schema and **outputs**: its corresponding GO structures code. Please note that this file does not need to be reviewed, since it is a copied file from an existing repository: [go-avro](github.com/elodina/go-avro)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,2 @@
+generate-schema:
+	go run schema/codegen/codegen.go --schema schema/block.elrond.avsc --out schema/schema.go
```
Comment (miiu96) [inline Makefile]: add empty line at the end of the file

Comment (mariusmihaic) [reply Makefile]: Done

File: schema/codegen/codegen.go [hunk 2]
```diff
@@ -0,0 +1,81 @@
+package main
```
Comment (AdoAdoAdo) [inline schema/codegen/codegen.go]: please remove the code generation logic from the repository, and simply use the binary (you can add in the readme info about how to generate the code from schema and how to download the code generation binary)

**Review by AdoAdoAdo — APPROVED** [review]

File: schema/codegen/codegen.go (hunk 2 above)
Comment (mariusmihaic) [reply schema/codegen/codegen.go]: Good point.
I applied your suggestion! :)

File: README.md [hunk 3]
```diff
@@ -1 +1,11 @@
 # covalent-indexer-go
+
+## Avro schema update
+In case you want to modify the existing avro schema, after you finish your changes, you need to re-generate the corresponding code, by:
+
+1. Get code generator binary
+```bash
+go install github.com/elodina/go-avro/codegen@latest
```
Comment (miiu96) [inline README.md:8]: go get instead of install

**Review by miiu96 — APPROVED** [review]

File: README.md (hunk 3 above)
Comment (miiu96) [reply README.md:8]: instead of `@latest` you can use https://github.com/elodina/go-avro/releases/tag/v0.1.0.0 this release

**Pull request merged**